invenio_oauthclient/contrib/cern_openid.py

```diff
@@ -370,26 +372,34 @@ def on_identity_changed(sender, identity):
         disconnect_identity(identity)
         return
 
+    logged_in_via_token = hasattr(current_user, 'login_via_oauth2') \
+        and getattr(current_user, 'login_via_oauth2')
+
     client_id = current_app.config["CERN_APP_OPENID_CREDENTIALS"][
         "consumer_key"
     ]
-    account = RemoteAccount.get(
+    remote_account = RemoteAccount.get(
         user_id=current_user.get_id(), client_id=client_id
     )
     roles = []
-    if account:
-        remote = find_remote_by_client_id(client_id)
-        resource = get_resource(remote)
+
+    if remote_account and not logged_in_via_token:
         refresh = current_app.config.get(
             "OAUTHCLIENT_CERN_OPENID_REFRESH_TIMEDELTA",
             OAUTHCLIENT_CERN_OPENID_REFRESH_TIMEDELTA,
         )
-
-        roles.extend(
-            account_roles_and_extra_data(
-                account, resource, refresh_timedelta=refresh
+        if refresh:
```

Comment:
Do you need this `if refresh`? I think it will be always set?

Reply:
No, If you set a False value on the config the refresh will be turned off, I did it to leave the functionality of refresh, I thought it might be handy for some use cases